### PR TITLE
Rename UnloadedToolchainContextKey to ToolchainContextKey.

### PR DESCRIPTION
**Pull request by katre**
Cleanup leading to toolchain transitions, #10523.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/skyframe/ConfiguredTargetFunction.java [hunk 1]
```diff
@@ -485,11 +485,11 @@ public SkyValue compute(SkyKey key, Environment env) throws ConfiguredTargetFunc
             configuration.getFragmentsMap().keySet(),
             BuildOptions.diffForReconstruction(defaultBuildOptions, toolchainOptions));
 
-    Map<String, UnloadedToolchainContextKey> unloadedToolchainContextKeys = new HashMap<>();
+    Map<String, ToolchainContextKey> unloadedToolchainContextKeys = new HashMap<>();
```
Comment (juliexxia) [inline src/main/java/com/google/devtools/build/lib/skyframe/ConfiguredTargetFunction.java:488]: Should these relevant var names be changed to, e.g., toolchainContextKeys? Does UnloadedToolchainContext still exist? Sorry if I'm missing context!

File: src/test/java/com/google/devtools/build/lib/analysis/util/BuildViewForTesting.java [hunk 2]
```diff
@@ -501,19 +501,19 @@ public RuleContext getRuleContextForTesting(
     SkyFunctionEnvironmentForTesting skyfunctionEnvironment =
         skyframeExecutor.getSkyFunctionEnvironmentForTesting(eventHandler);
 
-    Map<String, UnloadedToolchainContextKey> unloadedToolchainContextKeys = new HashMap<>();
+    Map<String, ToolchainContextKey> unloadedToolchainContextKeys = new HashMap<>();
```
Comment (juliexxia) [inline src/test/java/com/google/devtools/build/lib/analysis/util/BuildViewForTesting.java:504]: same renaming comment here and below

File: src/main/java/com/google/devtools/build/lib/skyframe/ConfiguredTargetFunction.java (hunk 1 above)
Comment (katre) [reply src/main/java/com/google/devtools/build/lib/skyframe/ConfiguredTargetFunction.java:488]: UnloadedToolchainContext still exists, but yes, these should be renamed.

File: src/test/java/com/google/devtools/build/lib/analysis/util/BuildViewForTesting.java (hunk 2 above)
Comment (katre) [reply src/test/java/com/google/devtools/build/lib/analysis/util/BuildViewForTesting.java]: Done.

**Pull request closed without merging**